src/main/java/org/opensearch/geospatial/action/upload/geojson/Uploader.java

```diff
@@ -139,4 +150,23 @@ private void indexContentAsDocument(
             uploadStepListener.onFailure(new IllegalStateException(message.toString()));
         }));
     }
+
+    private void injectPostUploadMetrics(BulkResponse response) {
+        int total = response.getItems().length;
+        metricBuilder.uploadCount(total);
+        metricBuilder.duration(response.getTook().duration());
+        if (response.hasFailures()) {
```

Comment:
!hasFailures?

Reply:
Good catch. Added test to validate the same.